internal/rbd/replicationcontrollerserver.go

```diff
@@ -570,24 +587,27 @@ func (rs *ReplicationServer) ResyncVolume(ctx context.Context,
 	// If the image state on both the sites are up+unknown consider that
 	// complete data is synced as the last snapshot
 	// gets exchanged between the clusters.
-	if state == upAndUnknown {
+	if state == unknown && localStatus.Up {
 		ready = true
-		for _, s := range mirrorStatus.PeerSites {
-			if imageMirroringState(s.State) != upAndUnknown {
-				util.UsefulLog(
-					ctx,
-					"peer site name=%s, mirroring state=%s, description=%s and lastUpdate=%s",
-					s.SiteName,
-					s.State,
-					s.Description,
-					s.LastUpdate)
-				ready = false
+		for _, s := range mirrorStatus.SiteStatuses {
```

Comment:
Consider moving this for loop to a helper function that returns the true/false you can assign to `ready`. This makes the function simpler and no need for the `// nolint` anymore

Reply:
sure.